File: src/main/java/com/example/demoAllTouhou/services/GameCharacterService.java
package com.example.demoAllTouhou.services;

import com.example.demoAllTouhou.entities.GameCharacterEntity;
import com.example.demoAllTouhou.repositories.GameCharacterRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GameCharacterService {

    @Autowired
    private GameCharacterRepository gameCharacterRepository;

    // **Crear un nuevo GameCharacter**
    public GameCharacterEntity createGameCharacter(GameCharacterEntity gameCharacterEntity) {
        return gameCharacterRepository.save(gameCharacterEntity);
    }

    // **Obtener todos los GameCharacters**
    public List<GameCharacterEntity> getAllGameCharacters() {
        return gameCharacterRepository.findAll();
    }

    // **Obtener un GameCharacter por ID**
    public GameCharacterEntity getGameCharacterById(Long id) {
        return gameCharacterRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("GameCharacter not found with id: " + id));
    }

    // **Actualizar un GameCharacter**
    public GameCharacterEntity updateGameCharacter(Long id, GameCharacterEntity updatedGameCharacter) {
        GameCharacterEntity existingGameCharacter = gameCharacterRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("GameCharacter not found with id: " + id));

        existingGameCharacter.setCharacterId(updatedGameCharacter.getCharacterId());
        existingGameCharacter.setGameId(updatedGameCharacter.getGameId());

        return gameCharacterRepository.save(existingGameCharacter);
    }

    // **Eliminar un GameCharacter por ID**
    public void deleteGameCharacter(Long id) {
        if (!gameCharacterRepository.existsById(id)) {
            throw new RuntimeException("GameCharacter not found with id: " + id);
        }
        gameCharacterRepository.deleteById(id);
    }

    // **Obtener todos los GameCharacters asociados a un Character ID**
    public List<GameCharacterEntity> getGameCharactersByCharacterId(Long characterId) {
        return gameCharacterRepository.findByCharacterId(characterId);
    }

    // **Obtener todos los GameCharacters asociados a un Game ID**
    public List<GameCharacterEntity> getGameCharactersByGameId(Long gameId) {
        return gameCharacterRepository.findByGameId(gameId);
    }
}

File: Frontend/src/services/GameCharacterService.js
import httpClient from "../http-common";

// Crea un nuevo GameCharacter
const createGameCharacter = (gameCharacter) => {
    return httpClient.post(`/game-characters`, gameCharacter);
};

// Obtiene todos los GameCharacters
const getAllGameCharacters = () => {
    return httpClient.get(`/game-characters`);
};

// Obtiene un GameCharacter por ID
const getGameCharacterById = (id) => {
    return httpClient.get(`/game-characters/${id}`);
};

// Actualiza un GameCharacter
const updateGameCharacter = (id, gameCharacter) => {
    return httpClient.put(`/game-characters/${id}`, gameCharacter);
};

// Elimina un GameCharacter por ID
const deleteGameCharacter = (id) => {
    return httpClient.delete(`/game-characters/${id}`);
};

// Obtiene todos los GameCharacters asociados a un Character ID
const getGameCharactersByCharacterId = (characterId) => {
    return httpClient.get(`/game-characters/by-character/${characterId}`);
};

// Obtiene todos los GameCharacters asociados a un Game ID
const getGameCharactersByGameId = (gameId) => {
    return httpClient.get(`/game-characters/by-game/${gameId}`);
};

export default {
    createGameCharacter,
    getAllGameCharacters,
    getGameCharacterById,
    updateGameCharacter,
    deleteGameCharacter,
    getGameCharactersByCharacterId,
    getGameCharactersByGameId
};
